src/progpy/mixture_of_experts.py

```diff
@@ -0,0 +1,213 @@
+# Copyright © 2021 United States Government as represented by the Administrator of the
+# National Aeronautics and Space Administration.  All Rights Reserved.
+
+import numpy as np
+
+from progpy import PrognosticsModel, CompositeModel
+
+DIVIDER = '.'
+
+
+class MixtureOfExpertsModel(CompositeModel):
+    """
+    .. versionadded:: 1.6.0
+
+    Mixture of Experts (MoE) models combine multiple models of the same system, similar to Ensemble models. Unlike Ensemble Models, the aggregation is done by selecting the "best" model. That is the model that has performed the best over the past.
+
+    The MoE model's inputs include the inputs and outputs of the individual models making up the model. If the output values are provided as an input to the model then the model will update the score during state tranisition. If not, state transition will continue as normal. Typically, outputs are provided in the MoE model input when performing a state estimation step but not when predicting forward.
+
+    Scores for the individual models is tracked in the state. At a state transition when outputs are provided, the score for the best model will increase by max_score_step for the best fitting model (i.e., lowest error in output) and decrease by max_score_step for the worst. All other models will be scaled between these, based on the error.
```

Comment:
```suggestion
    Mixture of Experts (MoE) models combine multiple models of the same system, similar to Ensemble models. Unlike Ensemble Models, the aggregation is done by selecting the "best" model. That is the model that has performed the best over the past. Each model will have a 'score' that is tracked in the state, and this determines which model is best. 

    The MoE model's inputs include the inputs and outputs of the individual models making up the model. If the output values are provided as an input to the model then the model will update the score during state transition. The outputs are required to update the score, since this allows for the calculation of how "good" the model is. If outputs are not provided, state transition will continue as normal, without updating model scores. 

    At a state transition when outputs are provided, the score for the best model will increase by max_score_step for the best fitting model (i.e., lowest error in output) and decrease by max_score_step for the worst. All other models will be scaled between these, based on the error.
    
    Typically, outputs are provided in the MoE model input when performing a state estimation step (i.e. when there is measured data) but not when predicting forward (i.e. when the output is unknown).
```